### PR TITLE
chore(doc): Design a no-result page

### PR DESCRIPTION
**Pull request by bobylito**
First part of #2103

### CODE REVIEW

**Comment by bobylito** [comment]
https://deploy-preview-2269--algolia-instantsearch.netlify.com/guides/no-result.html

File: docgen/src/guides/no-result.md [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+withHeadings: true
+navWeight: 10
+editable: true
+githubSource: docgen/src/guides/no-result.md
+---
+## Designing the no results page
+
+When the query returns no results, it is important to let the user know that their query led to no results.
+
+By doing so, we’re acknowledging that not all queries lead to some result and with some additional hints we can let them continue to use the application. This way, there is less chance that the user will leave the website to do a search to an external search engine.
```
Comment (iam4x) [inline docgen/src/guides/no-result.md]: continue application -> continue search?

File: docgen/src/guides/no-result.md [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+There are a few strategies that can be implemented from the very simple to some more complex strategies. This guide will walk you through one that can be easily implemented with InstantSearch.js: we’ll first improve the message that we provide to the user and then we’ll add a button to let the user clear the filters.
```
Comment (iam4x) [inline docgen/src/guides/no-result.md]: Sentence is a bit long, maybe divide it in two:

```md
There are a few strategies that can be implemented from the very simple to some more complex strategies. 
This guide will walk you through one that can be easily implemented with InstantSearch.js:

* We’ll first improve the message that we provide to the user 
* And then we’ll add a button to let the user clear the filters.
```

File: docgen/src/guides/no-result.md [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+By default, InstantSearch.js will only display "no results" when there are no results. The bare minimum to handle the no-result case is to provide the user with a message that indicates that no results were found in a friendly fashion.
+
+In order to do that, we can use the InstantSeach.js’ Hits widget option [templates.empty](https://community.algolia.com/instantsearch.js/v2/widgets/hits.html#struct-HitsTemplates-empty):
+
+```javascript
+const search = instantsearch(/* parameters */);
+search.addWidget(instantsearch.widgets.hits({
+  container: ‘hits’,
+  templates: {
+    empty: ‘<p class="info”>No results were found with your current filters. Try to remove some filters or change the search query.</p>’,
```
Comment (iam4x) [inline docgen/src/guides/no-result.md]: It should be `<p class="info">` and not this weird `”` character from google docs.

File: docgen/src/guides/no-result.md [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+URL sync makes your InstantSearch.js app aware of changes in the URL. With this, we can easily influence the parameters of the search. In this case, we want to clear all the filters so we can make a link to this page, without the parameters.
+
+We do this by customizing again the no-result template and adding it a clear all link:
+
+```javascript
+const search = instantsearch(/* parameters */);
+search.addWidget(instantsearch.widgets.hits({
+  container: ‘hits’,
+  templates: {
+    empty: ‘<p class="info”>No results were found with your current filters. <br/> <a class=”button” href=”.”>Clear all the filters</a></p>’,
```
Comment (iam4x) [inline docgen/src/guides/no-result.md]: Same weird character `”`

**Comment by iam4x** [comment]
All good 🙌

File: docgen/src/guides/no-result.md [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+layout: main.pug
+category: guides
+withHeadings: true
+navWeight: 10
+editable: true
+githubSource: docgen/src/guides/no-result.md
+---
+## Designing the no results page
+
+When the query returns no results, it is important to let the user know that their query led to no results.
```
Comment (Haroenv) [inline docgen/src/guides/no-result.md]: When _a_ query

File: docgen/src/guides/no-result.md [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+search.addWidget(instantsearch.widgets.hits({
+  container: 'hits',
+  templates: {
+    empty: '<p class="info">No results were found with your current filters. Try to remove some filters or change the search query.</p>',
+    item: yourItemTemplate
+  },
+}));
+```
+
+When there is no results, the user will see a paragraph that says: "No results were found with your current filters. Try to remove some filters or change the search query.".
```
Comment (Haroenv) [inline docgen/src/guides/no-result.md]: _are_ no results

**Pull request merged**